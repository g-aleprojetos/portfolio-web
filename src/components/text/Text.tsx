import React from 'react';

import {TextStyled, TextProps} from './Text.styles';

export function Text(props: TextProps) {
  return (
    <TextStyled
      $tamanho={props.tamanho}
      $alturaDeLinha={props.alturaDeLinha}
      $marginTop={props.marginTop}
      $toUpper={props.toUpper}
      {...props}
    />
  );
}
