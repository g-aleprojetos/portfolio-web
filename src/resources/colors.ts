export const colors = {
  white: '#ffffff',
  lightgray: '#d3d3d3',
  black: '#000000',
  black50per: '#00000080',
  darkgray: '#23242a',
  yellow: '#FFE879',
  blue: '#15154E',
  lightblue: '#2196F3',
};
