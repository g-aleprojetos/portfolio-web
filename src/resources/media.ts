const devices = {
  xmobile: 375,
  mobile: 425,
  tablet: 768,
  laptop: 1023,
  xlaptop: 1440,
};

export const media = {
  xmobile: `(max-width: ${devices.xmobile}px)`,
  mobile: `(max-width: ${devices.mobile}px)`,
  tablet: `(max-width: ${devices.tablet}px)`,
  laptop: `(max-width: ${devices.laptop}px)`,
  xlaptop: `(max-width: ${devices.xlaptop}px)`,
};
