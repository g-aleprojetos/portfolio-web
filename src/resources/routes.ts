export default {
  Home: '/',
  Projects: '/projects',
  About: '/about',
  Settings: '/settings',
};
