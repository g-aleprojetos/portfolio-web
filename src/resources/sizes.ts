export default {
  px1: '0.063rem',
  px2: '0.125rem',
  px3: '0.188rem',
  px5: '0.313rem',
  px8: '0.5rem',
  px13: '0.813rem',
  px21: '1.313rem',
  px34: '2.125rem',
  px55: '3.438rem',
  px89: '5.563rem',
  px144: '9rem',
  px233: '14.563rem',
  px377: '23.563rem',
  px610: '38.125rem',
  media_768: '768px',
};
