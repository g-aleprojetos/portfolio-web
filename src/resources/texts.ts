const tamanho = {
  xxsmall: 12,
  xsmall: 14,
  small: 16,
  medium: 18,
  big: 24,
};

const altura = {
  xxsmall: 18,
  xsmall: 22,
  small: 24,
  medium: 26,
  big: 34,
};

export default {tamanho, altura};
