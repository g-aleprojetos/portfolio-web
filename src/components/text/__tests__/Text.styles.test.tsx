import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import {TextStyled} from '../Text.styles';
import {colors} from 'resources/colors';

describe('Texto.styles', () => {
  test('DEVE ser igual ao snapshot', () => {
    const {container} = render(<TextStyled>Teste</TextStyled>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test(`DEVE ter a cor igual a "##000000" QUANDO a propriedade cor informada for ${colors.black} `, () => {
    const {container} = render(
      <TextStyled color={colors.black}>Teste</TextStyled>,
    );
    expect(container.firstChild).toHaveStyleRule('color', '#000000');
  });

  test('DEVE ter o tamanho da fonte igual a "14" QUANDO a propriedade tamanho for igual a "14"', () => {
    const {container} = render(<TextStyled $tamanho={14}>Teste</TextStyled>);
    expect(container.firstChild).toHaveStyleRule('font-size', '14px');
  });

  test('DEVE ter a marginTop igual a "10" QUANDO a propriedade marginTop for igual a "10"', () => {
    const {container} = render(<TextStyled $marginTop={10}>Teste</TextStyled>);
    expect(container.firstChild).toHaveStyleRule('margin-top', '10px');
  });

  test('DEVE ter o fontFamily igual a "avertaSemibold" QUANDO a propriedade peso for igual a "bold"', () => {
    const {container} = render(<TextStyled peso="bold">Teste</TextStyled>);
    expect(container.firstChild).toHaveStyleRule(
      'font-family',
      'AvertaStd-Bold',
    );
  });

  test('DEVE ter o text-transform igual a "uppercase" QUANDO a propriedade toUpper for igual a "true"', () => {
    const {container} = render(<TextStyled $toUpper={true}>Teste</TextStyled>);
    expect(container.firstChild).toHaveStyleRule('text-transform', 'uppercase');
  });

  test('DEVE ter o cursor igual a "pointer" QUANDO a propriedade cursor for igual a "pointer"', () => {
    const {container} = render(<TextStyled cursor="pointer">Teste</TextStyled>);
    expect(container.firstChild).toHaveStyleRule('cursor', 'pointer');
  });

  test('DEVE ter o line-height igual a "20" QUANDO a propriedade for passada no alturaDeLinha', () => {
    const {container} = render(
      <TextStyled $alturaDeLinha={20}>Teste</TextStyled>,
    );
    expect(container.firstChild).toHaveStyleRule('line-height', '20px');
  });
});
