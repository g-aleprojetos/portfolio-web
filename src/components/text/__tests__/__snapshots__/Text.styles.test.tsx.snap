// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Texto.styles DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #ffffff;
  margin-top: 0px;
  text-transform: none;
  cursor: text;
}

<p
  class="c0"
>
  Teste
</p>
`;
